reference-implementation/lib/writable-stream.js

```diff
@@ -4,225 +4,661 @@ const { InvokeOrNoop, PromiseInvokeOrNoop, PromiseInvokeOrFallbackOrNoop, Valida
         typeIsObject } = require('./helpers.js');
 const { rethrowAssertionErrorRejection } = require('./utils.js');
 const { DequeueValue, EnqueueValueWithSize, GetTotalQueueSize, PeekQueueValue } = require('./queue-with-sizes.js');
-const CountQueuingStrategy = require('./count-queuing-strategy.js');
 
 class WritableStream {
-  constructor(underlyingSink = {}, { size, highWaterMark = 0 } = {}) {
-    this._underlyingSink = underlyingSink;
+  constructor(underlyingSink = {}, { size, highWaterMark } = {}) {
+    // Temporary value. Never used. To be overwritten by the initializer code of the controller.
+    this._state = 'writable';
+    this._storedError = undefined;
 
-    this._closedPromise = new Promise((resolve, reject) => {
-      this._closedPromise_resolve = resolve;
-      this._closedPromise_reject = reject;
-    });
+    this._writer = undefined;
 
-    this._readyPromise = Promise.resolve(undefined);
-    this._readyPromise_resolve = null;
+    this._writeRequests = [];
```

Comment:
Fixed by https://github.com/whatwg/streams/pull/488/commits/e57e25e9385d2ce243e7da0de4b95d9f35d9cbcb

I'm not sure if it's what you meant. Please take a look again.